src/Utilities/NumberUtil.php

```diff
@@ -0,0 +1,34 @@
+<?php
+/**
+ * A class of utilities for dealing with numbers.
+ */
+
+namespace Automattic\WooCommerce\Utilities;
+
+/**
+ * A class of utilities for dealing with numbers.
+ */
+final class NumberUtil {
+
+	/**
+	 * Round a number using the built-in `round` function if the value to round is numeric
+	 * (a number or a string that can be parsed as a number), or 0 otherwise.
+	 *
+	 * This is needed because in PHP 7 applying `round` to a non-numeric value returns 0,
+	 * but in PHP 8 it throws an error. Specifically, in WooCommerce we have a few places where
+	 * round('') is often executed.
+	 *
+	 * @param mixed $val The value to round.
+	 * @param int   $precision The optional number of decimal digits to round to.
+	 * @param int   $mode A constant to specify the mode in which rounding occurs.
+	 *
+	 * @return float The value rounded to the given precision as a float, or the supplied default value.
+	 */
+	public static function round( $val, int $precision = 0, int $mode = PHP_ROUND_HALF_UP ) : float {
+		if ( is_numeric( $val ) ) {
```

Comment:
there is a backward incompatibility here in that `is_numeric` will also reject strings when there are spaces or characters around a valid numeric value. See in PHP 7.2:
```sh
>>> $number_with_spaces = ' 1.5 ';
=> " 1.5 "
>>> NumberUtil::round( $number_with_spaces )
=> 0.0
>>> round( $number_with_spaces )
=> 2.0
>>>
```

perhaps using something like `floatval` before passing to round will be better? wdyt?

Reply:
Good point. I'll give that a try, thanks.

Reply:
@vedanshujain I've fixed it, now numbers with spaces work as expected. Also fixed a couple more of cases to act like the built-in `round`:

- For the boolean value `true` it now returns 1, not 0.
- For objects it now throws an error (that was the case in PHP 7 already), instead of returning 0.

Reply:
Looks good. Thanks!